Modules/Product/Http/Services/ProductService.php

```diff
@@ -103,12 +110,114 @@ public function fetch($request): array
 
         $data = array(
             "products" => $resultArray,
-            "pblshprdcts_count" => $publishCount,
-            "unpblshprdcts_count" => $unpublishedCount,
-            "allprdcts_count" => $productsCount
+            "pblshprdcts_count" => $publishedProductsCount,
+            "unpblshprdcts_count" => $unpublishedProductsCount,
+            "allprdcts_count" => $allProductsCount
         );
 
 
+        return ['res' => true, 'msg' => "", 'data' => $data];
+    }
+
+    /**
+     * Fetch All products By logged brand
+     *
+     * @return array
+     */
+
+    public function getProducts(): array
+    {
+        $userId = auth()->user()->id;
+        $data = [];
+        $products = Product::where('user_id', $userId)->where('status', 'publish')->get();
+        foreach ($products as $v) {
+            $productStock = $v->stock;
+            $productVariations = Product::find($v->id)->productVariation()->where('status', '1')->where('stock','>', '0')->get();
+            $productVariationsCount = $productVariations->count();
+            if ($productVariationsCount > 0) {
+                $variantMinPrice = Product::find($v->id)->productVariation()->min('price');
+                $price = $variantMinPrice . '+';
+                $variantStock = Product::find($v->id)->productVariation()->sum('stock');
+                $availability = $variantStock > 0 ? 'in stock' : 'out of stock';
+                $productStock += $variantStock;
+            } else {
+                $price = $v->usd_wholesale_price;
+                $availability = $v->stock > 0 ? 'in stock' : 'out of stock';
+            }
```

Comment:
Optimize this query in same way like the above one, get the relation productVariation with eager loading while query products like Product::with('productVariation').....